### PR TITLE
Clarify what's left in handling OP_MSG checksum

### LINKED ISSUE #2691: Clarify what's left in handling OP_MSG checksum

### PR DESCRIPTION
**Pull request by rumyantseva**
# Description

Closes #2691.

I created a new issue #2690 to address the current challenges around checksum calculation. 

## Readiness checklist

* [ ] I added/updated unit tests.
* [ ] I added/updated integration/compatibility tests.
* [x] I added/updated comments and checked rendering.
* [x] I made spot refactorings.
* [ ] I updated user documentation.
* [x] I ran `task all`, and it passed.
* [x] I ensured that PR title is good enough for the changelog.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

File: internal/wire/msg_body.go [hunk 1]
```diff
@@ -150,34 +151,42 @@ func getChecksum(data []byte) (uint32, error) {
...
+	}
 
-		got := hasher.Sum32()
-		if want != got {
-			return lazyerrors.New("OP_MSG checksum does not match contents.")
-		}
+	offset := len(body) - crc32.Size
+	if err := binary.Write(hasher, binary.LittleEndian, body[:offset]); err != nil {
+		return lazyerrors.Error(err)
+	}
```
Comment (chilagrow) [inline internal/wire/msg_body.go:185]: > To write the checksum, we need to modify the header by summing four bytes in the message length field for the checksum.

Does the quote from https://github.com/FerretDB/FerretDB/issues/2690 refer above two statements? Just checking :hugs:

**Review by chilagrow — APPROVED** [review]
Looks good to me. Thanks for the detailed issue

**Review by noisersup — APPROVED** [review]
Looks good to me!

File: internal/wire/msg_body.go (hunk 1 above)
Comment (rumyantseva) [reply internal/wire/msg_body.go:185]: The above two statements are to calculate the checksum. To calculate checksum we need the whole header plus the whole body except the last four bytes.

**Pull request merged**